core/filters/browser.rb

```diff
@@ -46,7 +46,7 @@ def self.is_valid_browserversion?(str)
       return false if has_non_printable_char?(str)
       return true if str.eql? 'UNKNOWN'
       return true if str.eql? 'ALL'
-      return false if !nums_only?(str) and !is_valid_float?(str)
+      return false if !nums_only?(str) and !str.match(/\b\d+(\.\d+)+\b/)
```

Comment:
Why is this change necessary? The browser version should not contain `\b`.

Reply:
Float doesn't take into account longer version numbers separated by dots. FF passes that with number that contains one dot (e. g. 130.4), however, C fails with multiple dots (e. g. 131.0.6738.0).

Without word boundaries, the regex becomes more permissive and can match sequences that are embedded within larger strings, even if that’s not what you want. Word boundaries (\b) are crucial when you need to ensure the version number (or similar pattern) is isolated and not part of a larger string or word.

Reply:
The suggested changes do not preserve the validation logic. These changes allow strings containing `\b` to be treated as valid browser versions.

The core filters are used for validation. This method is used to validate the browser version in multiple areas of the application.

The current implementation first checks for `ALL` and `UNKNOWN` (this is due to poor design). It then rejects any values which are not numbers or floats. While the float check is clearly an oversight as `major.minor.revision` strings should be supported (`/[0-9\.]+/`), the suggested approach will match other values such as `test 1.2.3.4 test` (so long as it is less than 20 characters) which is not a valid browser version.

```ruby
#!/usr/bin/env ruby

def valid?(str)
  str.match(/\b\d+(\.\d+)+\b/) ? true : false
end

puts valid?('test 1.2.3.4 test')

```

You probably want to do something like this:

```suggestion
      return false unless str.match(/\A\d+([\d\.]+)\z/)
```

Although this approach is also flawed in that browser versions can feasibly contain letters. `1.2.3a` or `1.2.3-alpha` for example.

A more robust long term approach would be to leverage the `Gem::Version` library and split the input string by `-` and ` `. However, this is outside the scope of this PR.

Reply:
Thanks, mate, for elaboration, good point, yours is better than mine due to the beginning and ending anchors rather than word boundaries. However, I've noticed a slight discrepancy in there, it allows for strings like 1.1.1. (ending with the dot). I'm about to commit a slightly modified version of it:  `str.match(\A\d+(\.\d+)+\z)` .